### PR TITLE
feat: show OG post of comment in detail page

### PR DESCRIPTION
**Pull request by sasicodes**
## What does this PR do?

Fixes # (issue)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Enhancement (non-breaking small changes to existing functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How should this be tested?

- [ ] Test A
- [ ] Test B

## Checklist

- I haven't read the [contributing guide](https://github.com/lensterxyz/lenster/blob/main/CONTRIBUTING.md)
- My code doesn't follow the style guidelines of this project
- I haven't performed a self-review of my own code and corrected any misspellings
- I haven't commented my code, particularly in hard-to-understand areas
- I haven't checked if my PR needs changes to the documentation
- I haven't checked if my changes generate no new warnings
- I haven't added tests that prove my fix is effective or that my feature works
- I haven't checked if new and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by sasicodes** [comment]
Blocked by CLS in embeds and images.

**Comment by bigint** [comment]
@sasicodes bls fix conflicts 🙇🏼

**Comment by bigint** [comment]
This is getting cutoff

image

**Comment by bigint** [comment]
Link T-4008

**Comment by sasicodes** [comment]
> This is getting cutoff
> 
> image

We have set the sidebar already 👍🏻 

Two option:
- It requires scroll area so that it will not hide.
- If we reduce the extra scroll, the sidebar bit hidden

**Comment by bigint** [comment]
@sasicodes It's ok to have scroll, one last thing can we handle for staff mode as well? using `useStaffMode` hook 🙇🏼 

image

File: apps/web/src/components/Publication/FullPublication.tsx [hunk 1]
```diff
@@ -36,10 +44,39 @@ const FullPublication: FC<Props> = ({ publication }) => {
     : publication?.stats?.totalAmountOfCollects;
   const showStats = mirrorCount > 0 || reactionCount > 0 || collectCount > 0;
 
+  const isGatedThread = !publication?.isGated && !commentOn?.isGated && !mainPost?.isGated;
+  // [...getURLs(mainPost?.metadata.content ?? ''), ...getURLs(commentOn?.metadata.content ?? '')].length ===
+  // 0;
```
Comment (bigint) [inline apps/web/src/components/Publication/FullPublication.tsx:49]: We can remove this comment 🗑️

File: apps/web/src/components/Publication/FullPublication.tsx [hunk 2]
```diff
@@ -4,26 +4,34 @@ import getAppName from '@lib/getAppName';
...
-const FullPublication: FC<Props> = ({ publication }) => {
+const FullPublication: FC<Props> = ({ publication, postContainerRef }) => {
+  const threadRef = useRef<HTMLDivElement>(null);
+
   const isMirror = publication.__typename === 'Mirror';
   const profile = isMirror ? publication?.mirrorOf?.profile : publication?.profile;
   const timestamp = isMirror ? publication?.mirrorOf?.createdAt : publication?.createdAt;
 
+  const commentOn = publication.__typename === 'Comment' ? (publication?.commentOn as any) : null;
+  const mainPost = commentOn && commentOn?.__typename !== 'Mirror' ? commentOn?.mainPost : null;
```
Comment (bigint) [inline apps/web/src/components/Publication/FullPublication.tsx:33]: Do we need `commentOn &&` condition? cuz the second one has optional chaining in it

File: apps/web/src/components/Publication/FullPublication.tsx [hunk 3]
```diff
@@ -36,10 +44,39 @@ const FullPublication: FC<Props> = ({ publication }) => {
...
 
+  const isGatedThread = !publication?.isGated && !commentOn?.isGated && !mainPost?.isGated;
+  // [...getURLs(mainPost?.metadata.content ?? ''), ...getURLs(commentOn?.metadata.content ?? '')].length ===
+  // 0;
+
+  const scrollToThread = () => {
+    if ((!mainPost && !commentOn) || !threadRef.current) {
+      return;
+    }
+    if (isGatedThread) {
```
Comment (bigint) [inline apps/web/src/components/Publication/FullPublication.tsx:55]:
```suggestion
    }

    if (isGatedThread) {
```

added space between two statements 🙇🏼

File: apps/web/src/components/Publication/FullPublication.tsx [hunk 4]
```diff
@@ -36,10 +44,39 @@ const FullPublication: FC<Props> = ({ publication }) => {
...
+  };
+
+  useLayoutEffect(() => {
+    scrollToThread();
+  });
+
+  // useLayoutEffect(() => {
+  //   scrollToThread();
+  //   // eslint-disable-next-line react-hooks/exhaustive-deps
+  // }, [threadRef, showOriginalThread]);
```
Comment (bigint) [inline apps/web/src/components/Publication/FullPublication.tsx:67]: can remove this

File: apps/web/src/components/Publication/index.tsx [hunk 5]
```diff
@@ -37,6 +42,24 @@ const ViewPublication: NextPage = () => {
     skip: !id
   });
 
+  const calculateAdaptiveHeight = () => {
+    const currentRef = postContainerRef.current;
+    if (!currentRef || !data?.publication) {
+      return;
+    }
+
+    // const availableLinks = getURLs(data?.publication?.metadata.content);
```
Comment (bigint) [inline apps/web/src/components/Publication/index.tsx:51]: can remove this

**Review by bigint — CHANGES_REQUESTED** [review]

**Comment by sasicodes** [comment]
Cleaned up 👍🏻

File: apps/web/src/components/UI/GridLayout.tsx [hunk 6]
```diff
@@ -4,6 +4,7 @@ interface Props {
   children: ReactNode;
   className?: string;
   classNameChild?: string;
+  style?: React.CSSProperties;
```
Comment (bigint) [inline apps/web/src/components/UI/GridLayout.tsx:7]: Just import `CSSProperties` from react 🙇🏼

**Review by bigint — APPROVED** [review]

File: apps/web/src/components/UI/GridLayout.tsx (hunk 6 above)
Comment (bigint) [reply apps/web/src/components/UI/GridLayout.tsx:7]: @sasicodes one last change

**Comment by bigint** [comment]
**LGTM** :shipit: 

Rad LGTM

File: apps/web/src/components/Publication/ThreadBody.tsx [hunk 7]
```diff
@@ -11,9 +11,10 @@ import PublicationHeader from './PublicationHeader';
 
 interface Props {
   publication: Publication;
+  intersectionRef?: any;
```
Comment (harish-sethuraman) [inline apps/web/src/components/Publication/ThreadBody.tsx:14]: Nice `any`

File: apps/web/src/components/UI/GridLayout.tsx [hunk 8]
```diff
@@ -18,6 +19,10 @@ export const GridItemFour: FC<Props> = ({ children, className = '' }) => {
   return <div className={`lg:col-span-4 md:col-span-12 col-span-12 ${className}`}>{children}</div>;
 };
 
-export const GridItemEight: FC<Props> = ({ children, className = '' }) => {
-  return <div className={`lg:col-span-8 md:col-span-12 col-span-12 mb-5 ${className}`}>{children}</div>;
+export const GridItemEight: FC<Props> = ({ children, className = '', style }) => {
+  return (
+    <div className={`lg:col-span-8 md:col-span-12 col-span-12 mb-5 ${className}`} style={style}>
```
Comment (harish-sethuraman) [inline apps/web/src/components/UI/GridLayout.tsx:24]:
```suggestion
    <div className={clsx('lg:col-span-8 md:col-span-12 col-span-12 mb-5', className)} style={style}>
```

**Comment by bigint** [comment]
@sasicodes is it good to go?

**Comment by sasicodes** [comment]
> @sasicodes is it good to go?

resolved conflicts

**Comment by bigint** [comment]
Closing this we will do this during redesign 🙇🏼

**Pull request closed without merging**